client/app/pages/dashboards/ShareDashboardDialog.jsx

```diff
@@ -0,0 +1,122 @@
+import { replace } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import Switch from 'antd/lib/switch';
+import Modal from 'antd/lib/modal';
+import Form from 'antd/lib/form';
+import Tooltip from 'antd/lib/tooltip';
+import { $http, toastr } from '@/services/ng';
+import { wrap as wrapDialog, DialogPropType } from '@/components/DialogWrapper';
+import InputWithCopy from '@/components/InputWithCopy';
+
+const API_SHARE_URL = 'api/dashboards/{id}/share';
+const HELP_URL = 'https://redash.io/help/user-guide/dashboards/sharing-dashboards?source=dialog';
+
+class ShareDashboardDialog extends React.Component {
+  static propTypes = {
+    dashboard: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+    dialog: DialogPropType.isRequired,
+  };
+
+  formItemProps = {
+    labelCol: { span: 8 },
+    wrapperCol: { span: 16 },
+    style: { marginBottom: 7 },
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      saving: false,
+      apiUrl: replace(API_SHARE_URL, '{id}', props.dashboard.id),
```

Comment:
After `supes(props)` call you can use `this.props`

Reply:
Considering `apiUrl` does not change in the component, wouldn't it be better for it to be a `getter` or a variable instead of a state?

Reply:
It's even better if it will be in a service (see https://github.com/getredash/redash/pull/3424#discussion_r255914090)

Reply:
@gabrieldutra a getter would execute the replace function each time it is accessed so I opted for either `this.state.apiUrl` or `this.apiUrl`. I'll swap it.